### PR TITLE
 Floyd’s Cycle-Finding Algorithm in Java

### PR DESCRIPTION
**Pull request by yashaswibiyahut**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

**YES**

### Description
_Issue no: #686_
**Approach:** Floyd’s Cycle-Finding Algorithm
This is the fastest method to solve this problem.
Time complexity: O(n).
Auxiliary Space:O(1).

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x]  I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)

### CODE REVIEW

**Comment by yashaswibiyahut** [comment]
kindly review this @Kajol-Kumari @SKAUL05 @siddharth25pandey @iamrajiv @HarshCasper

**Comment by yashaswibiyahut** [comment]
kindly review this @Kajol-Kumari @SKAUL05 @siddharth25pandey @iamrajiv @HarshCasper

File: Java/ds/DetectLoopLL.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+{
+    int data;
+    Node next;
+    Node(int x)
+    {
+        data = x;
+        next = null;
+    }
+}
+class DetectLoopLL
```
Comment (siddharth25pandey) [inline Java/ds/DetectLoopLL.java]: Add access modifier to the main class

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@yashaswibiyahut Do a small change, rest is fine

File: Java/ds/DetectLoopLL.java (hunk 1 above)
Comment (yashaswibiyahut) [reply Java/ds/DetectLoopLL.java]: Updated @siddharth25pandey  kindly review it.

**Review by iamrajiv — APPROVED** [review]

**Review by Kajol-Kumari — APPROVED** [review]
LGTM :+1:

**Comment by yashaswibiyahut** [comment]
kindly review this PR @siddharth25pandey  i made all the changes as per your requirements.

**Review by siddharth25pandey — APPROVED** [review]
@yashaswibiyahut Looks fine to me
Able to merge
Thanks!

**Pull request merged**